src/beliefpropagation/beliefpropagation.jl

```diff
@@ -1,104 +1,67 @@
 function message_tensors(
-  tn::ITensorNetwork;
-  nvertices_per_partition=nothing,
-  npartitions=nothing,
-  subgraph_vertices=nothing,
-  kwargs...,
+  ptn::PartitionedGraph; itensor_constructor=inds_e -> ITensor[dense(delta(inds_e))]
```

Comment:
I would separate this out into a `default_message_tensor_constructor(inds_e) = ...` function.

Also, probably should use `denseblocks` instead of `dense` to handle QN tensors.